pkg/web/httpservers.go

```diff
@@ -88,6 +93,10 @@ func NewHttpServer(
 		address: address,
 	}
 
+	if _, enablePromStatser := statser.(*stats.PrometheusStatser); enablePromStatser {
```

Comment:
There should also be an explicit `enablePromStatser` argument, otherwise the prom statser will be exposed to all http servers, which may not be what is desired.

```
if enablePromStatser {
  if _, ok := statser.(*stats.PrometheusStatser); ok {
    // TODO: append the route
  } else {
    return errNotPromStatser
  }
}
```